client/modules/IDE/components/Editor/index.jsx

```diff
@@ -81,36 +76,128 @@ window.HTMLHint = HTMLHint;
 
 const INDENTATION_AMOUNT = 2;
 
-class Editor extends React.Component {
-  constructor(props) {
-    super(props);
-    this.tidyCode = this.tidyCode.bind(this);
+function Editor(props) {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  // Defining states to usee
+  const files = useSelector((state) => state.files);
+  const ide = useSelector((state) => state.ide);
+  const file = useSelector(selectActiveFile);
+  const project = useSelector((state) => state.project);
+  const preferences = useSelector((state) => state.preferences);
+  const editorAccessibility = useSelector((state) => state.editorAccessibility);
+  const sidebarIsExpanded = useSelector((state) => state.ide.sidebarIsExpanded);
+  const consoleEvents = useSelector((state) => state.console);
+  const mergedState = {
+    ...preferences,
+    ...ide,
+    ...project,
+    ...editorAccessibility,
+    isExpanded: sidebarIsExpanded
+  };
+  // Binding Action Creators
+  const funcprop = bindActionCreators(
+    Object.assign(
+      {},
+      EditorAccessibilityActions,
+      FileActions,
+      ProjectActions,
+      IDEActions,
+      PreferencesActions,
+      UserActions,
+      ConsoleActions
+    ),
+    dispatch
+  );
+  // refs for codemirror
+  const cm = useRef(null);
+  // defining functions
+  function getFileMode(fileName) {
+    let mode;
+    if (fileName.match(/.+\.js$/i)) {
+      mode = 'javascript';
+    } else if (fileName.match(/.+\.css$/i)) {
+      mode = 'css';
+    } else if (fileName.match(/.+\.(html|xml)$/i)) {
+      mode = 'htmlmixed';
+    } else if (fileName.match(/.+\.json$/i)) {
+      mode = 'application/json';
+    } else if (fileName.match(/.+\.(frag|glsl)$/i)) {
+      mode = 'x-shader/x-fragment';
+    } else if (fileName.match(/.+\.(vert|stl)$/i)) {
+      mode = 'x-shader/x-vertex';
+    } else {
+      mode = 'text/plain';
+    }
+    return mode;
+  }
```

Comment:
I would move this function outside of the component since it does not use any variables of the component.